### DISCUSSION TITLE
Parsing partial code

### DISCUSSION THREAD

**Discussion by satbekmyrza** [tree-sitter/tree-sitter#1981] [Q&A] [answered]
Is it possible to parse a partial code using tree-sitter? 

Let's say we have a `binary_expression`
```
input <= 0.0 || output <= 0.0
```

Is it possible to parse just this code and build an AST with its root being a `binary_expression` node, not a `translation_unit` node? 

The above example would result in a valid 'partial' AST:
```
binary_expression
  left: binary_expression
    left: identifier
    right: number_literal
  right: binary_expression
    left: identifier
    right: number_literal
```

**Comment by ahelwer** [comment, marked as answer]
The search term you're looking for  in this repo's issues is "multiple entry points" and no it is not yet available.

**Reply by satbekmyrza** [reply]
Yes, that's exactly what I was looking for. Thank you for providing the search term! Here is the relevant issue #870.
